logprep/processor/domain_resolver/processor.py

```diff
@@ -134,70 +134,64 @@ def _tld_extractor(self):
-                            event_dbg = event["resolved_ip_debug"]
-                            if requires_storing:
-                                event_dbg["obtained_from_cache"] = False
-                            else:
-                                event_dbg["obtained_from_cache"] = True
-                            event_dbg["cache_size"] = len(self._domain_ip_map.keys())
-
-                        if self._domain_ip_map[hash_string] is not None:
-                            adding_was_successful = add_field_to(
-                                event, output_field, self._domain_ip_map[hash_string]
-                            )
-
-                            if not adding_was_successful:
-                                message = (
-                                    f"DomainResolver ({self.name}): The following "
-                                    f"fields already existed and "
-                                    f"were not overwritten by the DomainResolver: "
-                                    f"{output_field}"
-                                )
-                                raise ProcessingWarning(message=message)
-                            self.ps.increment_nested(self.name, "resolved_cache")
-                    except (context.TimeoutError, OSError):
-                        self._domain_ip_map[hash_string] = None
-                        self.ps.increment_nested(self.name, "timeouts")
-                    except UnicodeError as error:
-                        raise DomainResolverError(
-                            self.name, f"{error} for domain '{domain}'"
-                        ) from error
-                else:
-                    if output_field not in event:
-                        try:
-                            result = self._thread_pool.apply_async(socket.gethostbyname, (domain,))
-                            event[output_field] = result.get(timeout=self._config.timeout)
-                        except (context.TimeoutError, OSError):
-                            pass
-                        except UnicodeError as error:
-                            error_msg = f"{error} for domain '{domain}'"
-                            raise DomainResolverError(self.name, error_msg) from error
+        domain_or_url_str = get_dotted_field_value(event, domain_or_url)
+        if not domain_or_url_str:
+            return
+        domain = self._tld_extractor(domain_or_url_str).fqdn
+        if not domain:
+            return
+        self.ps.increment_nested(self.name, "total_urls")
+        if self._config.cache_enabled:
+            hash_string = self._hasher.hash_str(domain, salt=self._config.hash_salt)
+            requires_storing = self._cache.requires_storing(hash_string)
+            resolved_ip = None
+            if requires_storing:
+                resolved_ip = self._resolve_ip(domain, hash_string)
+                self._domain_ip_map.update({hash_string: resolved_ip})
```

Comment:
Why are we storing the resolved ips in a separate dict? Isn't this the job of the Cache? After all, the cache is an OrderedDict so we could use this to store the values and delete the `_domain_ip_map` entirely.

Reply:
no... in the cache there are only the hashes cause the cache is used in other modules too.